### ISSUE TITLE
readme states it executes 'cordova build' but actually executes 'cordova prepare' and 'cordova build'

### ISSUE THREAD

**Issue by josh-m-sharpe** [ionic-zone/fastlane-plugin-ionic#34]
See README: `This fastlane plugin helps you build your Ionic Cordova project via the ionic CLI and its ionic cordova build command.`

See code: https://github.com/ionic-zone/fastlane-plugin-ionic/blob/master/lib/fastlane/plugin/ionic/actions/ionic_action.rb#L89-L109

This is a problem because cordova's `build.json` is ignored by those two commands which has uses, like setting build flags:
https://github.com/apache/cordova-ios/issues/407

**Comment by janpio** [comment]
`cordova build` is just `cordova prepare` and then `cordova compile` stringed together: https://cordova.apache.org/docs/en/8.x/reference/cordova-cli/index.html#cordova-build-command

What you refer to, is probably taken care of by https://github.com/ionic-zone/fastlane-plugin-ionic/pull/31 (Correct?)

**Comment by josh-m-sharpe** [comment]
Possibly.  It seems that cordova prepare/compile don't respect options set in build.json... which does things other than just set buildFlags.

Why does this plugin use prepare and compile instead of just build?

**Comment by janpio** [comment]
1. Because the plugin this was forked off also did/does that ;)
2. It enables you to _not_ run `prepare` if you want more control about what is going on. (You could use the "blame" functionality of GitHub to find out the PR that added the functionality and read about the reasoning there)

**Comment by janpio** [comment]
> Ok. It's still incorrect?

No. `build` is just `prepare` and `compile` run after each other, so it is technically not incorrect. But I clarified in the README which commands it really runs: https://github.com/ionic-zone/fastlane-plugin-ionic/commit/77a01a7262d8ef552cecbdbc1ab70da994873300

> ionic's support for build.json provides value. Should this plugin allow its use?

This plugin just executes the Ionic CLI, so everything the CLI supports, this plugin also supports.

What _exactly_ are you doing and what is not working? iOS or Android? Can you share your project on GitHub?
